beets/dbcore/query.py

```diff
@@ -177,6 +177,23 @@ def string_match(cls, pattern, value):
         raise NotImplementedError()
 
 
+class StringQuery(StringFieldQuery):
+    """A query that matches a whole string in a specific item field."""
+
+    def col_clause(self):
+        search = (self.pattern
+                  .replace('\\', '\\\\')
+                  .replace('%', '\\%')
+                  .replace('_', '\\_'))
+        clause = self.field + " like ? escape '\\'"
```

Comment:
Could you elaborate a little bit on why this uses SQLite's `LIKE` operator instead of just plain `=`? Maybe I'm missing something, but it seems like that should work without any escaping…

Reply:
I am using `LIKE` to perform a case-insensitive string match, though you are right that there are several ways to achieve that which I can think of:

1. `$field LIKE $value`
    - Requires value to be escaped when building the query
    - Match can be satisfied by a `COLLATE NOCASE` index
2. `$field = $value COLLATE NOCASE`
    - Can be applied on a per-clause basis, so `WHERE artist = 'braid' COLLATE NOCASE AND album = 'No Coast'` works as expected
    - Match can be satisfied by a `COLLATE NOCASE` index
3. `UPPER($field) = UPPER($value)`.
    - Simple to understand
    - Not sure how to get sqlite to use an index for this match

Seems like sqlite does not understand how to change the case of non-ascii characters out of the box, so none of these methods will do the right thing for those strings... not sure if this is a deal-breaker?  If so, I am sure we can come up with some hack that will work reasonably well.

Reply:
Ohhhh, sorry for misunderstanding! I got it backward and thought this was the case-sensitive version; of course, that's just plain old `MatchQuery` in this PR. In that case, you're absolutely right and I don't think there's a strong reason to prefer either of the first two (good point about the index for option 3).

But in that case, is it a bug that `string_match` uses `pattern == value`? It should perhaps use `pattern.lower() == value.lower()` or similar for a similar effect in slow queries.

Reply:
I think we will just have to live with the consequences for non-ASCII characters. This is already the case for `SubstringQuery` for similar reasons. It's not great, but the complexity of working around it is also not terribly attractive.

Reply:
Yep, good call on `string_match` being screwed up.  Fixed that (and the test I added which was asserting the incorrect behavior...) in a [second commit](https://github.com/rcrowell/beets/commit/2cab2d670aa011006f4322a59176ba3dbb6bb22b) just now.

Looks like I was the one who got it backwards!